src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -241,6 +241,30 @@ static void window_track_place_update(rct_window* w)
             window_close(w);
 }
 
+static GameActionResult::Ptr FindValidTrackDesignPlaceHeight(CoordsXYZ& loc, uint32_t flags)
+{
+    for (int32_t i = 0; i < 7; i++)
+    {
+        auto tdAction = TrackDesignAction(CoordsXYZD{ loc.x, loc.y, loc.z, _currentTrackPieceDirection }, *_trackDesign);
```

Comment:
`CoordsXYZD` should have a `constexpr CoordsXYZD(CoordsXYZ _c, Direction _d)` ctor, it would make this clearer.